src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs

```diff
@@ -23,21 +23,31 @@ public static class VsHierarchyUtility
         public static string GetProjectPath(IVsHierarchy project)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
-
-            project.GetCanonicalName(VSConstants.VSITEMID_ROOT, out string projectPath);
+            var format = project as IPersistFileFormat;
+            format.GetCurFile(out string projectPath, out uint _);
```

Comment:
Do we need any defensive code in case the cast fails?  Why not use `(IPersistFileFormat)` rather than `as`? Although I have no idea if COM object typecast failures provide meaningful error messages, like .NET typecasts do.

Reply:
hmm, what if it does? It fails to initialize the project whose path we can't get :D 

We have a try/catch that'd log something in the output window, but check for null probably doesn't make it much more actionable.

I think this cast is always supposed to work.

Reply:
Sure, I'll update that.